### ISSUE TITLE
pod spec lint error with On Demand Resources

### ISSUE THREAD

**Issue by penelopearaujo** [CocoaPods/CocoaPods#10933]
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

- added a `.ttf` file to my project
- added a `on demand resource tag` to my `.ttf` file
- added `spec.on_demand_resources` key in my podspec and specified the path to the `.ttf` file
- ran `pod spec lint`

## What did you expect to happen?

expected the spec to pass validation so that I could push it to cocoapods

## What happened instead?

running `pod spec lint` resulted in the error:
 ```
- NOTE  | [iOS] xcodebuild:  error: On-Demand Resources is enabled (ENABLE_ON_DEMAND_RESOURCES = YES), but the PRODUCT_BUNDLE_IDENTIFIER build setting is empty (in target 'App' from project 'App')
```
I do not have a target or project called App, and the PRODUCT_BUNDLE_IDENTIFIER build setting is set in my project's info.plist as `$(PRODUCT_BUNDLE_IDENTIFIER)`

### Stack

```
   CocoaPods : 1.11.0
        Ruby : ruby 2.6.3p62 (2019-04-16 revision 67580) [universal.x86_64-darwin19]
    RubyGems : 3.0.3
        Host : Mac OS X 10.15.7 (19H1030)
       Xcode : 12.4 (12D4e)
         Git : git version 2.24.3 (Apple Git-128)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib
Repositories : master - git - https://github.com/CocoaPods/Specs.git @ a8db5c55f0110d6aa92bda70707e7b0e925a6624

               trunk - CDN - https://cdn.cocoapods.org/
```

### Installation Source

```
Executable Path: /usr/local/Cellar/cocoapods/1.11.0/libexec/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.5
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.1
cocoapods-trunk       : 1.6.0
cocoapods-try         : 1.2.0
```

**Comment by dnkoutso** [comment]
Oh that is a great issue report! We will fix for 1.11.1. We have apps building for ODR but no lint tests. Thanks a lot. We can ship this release hopefully this week.

**Comment by dnkoutso** [comment]
@penelopearaujo I cannot seem to reproduce locally. Do you have a sample repo I can try this with or a podspec?

**Comment by dnkoutso** [comment]
makes me wonder if this an Xcode related issue? I tried Xcode 13 beta and it passed and Xcode 12.5.1 and it passed locally.

**Comment by penelopearaujo** [comment]
Thanks, @dnkoutso! I will try again after updating Xcode to 12.5.1 version :) 

You can reproduce the error with [this sample project](https://github.com/penelopearaujo/palettes), which does nothing, but tests this ODR thing.

**Comment by dnkoutso** [comment]
awesome

**Comment by dnkoutso** [comment]
Yes I tried your sample repo and I cant repro at this time.

**Comment by dnkoutso** [comment]
@penelopearaujo in your repo there is no actual `.ttf` resource present?

**Comment by dnkoutso** [comment]
OK after messing around with it I repro'd! 

```
    - NOTE  | [iOS] xcodebuild:  /var/folders/yt/vszldnq143dgzjmmgr_tqs540000gn/T/CocoaPods-Lint-20210909-58421-1jay8rt-Palettes/App.xcodeproj: error: On-Demand Resources is enabled (ENABLE_ON_DEMAND_RESOURCES = YES), but the PRODUCT_BUNDLE_IDENTIFIER build setting is empty (in target 'App' from project 'App')
```

**Comment by penelopearaujo** [comment]
Yes! I forgot to commit after moving things around 🤦🏻‍♀️ 
Anyways, I'll let you know if it works after updating Xcode!

**Comment by dnkoutso** [comment]
it should not be an Xcode issue I dont think. I got a fix already coming up.

**Comment by penelopearaujo** [comment]
Cool! Thanks ;)

**Comment by dnkoutso** [comment]
PR https://github.com/CocoaPods/CocoaPods/pull/10936

**Comment by dnkoutso** [comment]
Fixed and merged! You can use bundler to point to the `1-11-stable` branch to get this fix, I will ship 1.11.1 this week or early next.

**Comment by penelopearaujo** [comment]
Hey! Updated the version (now using `1.11.2`), the podspec now passes validation, but `pod trunk push` returns error: 
```
[!] The Pod Specification did not pass validation.
The following validation failed:
- Warnings: Unrecognized `on_demand_resources` key.
```

Is it related or should I open another issue for this?
Thanks!

**Comment by dnkoutso** [comment]
do not open yet a new issue let me verify on my end.

**Comment by dnkoutso** [comment]
@penelopearaujo I just tried your sample repo again with 1.11.2 and it passes:

```
 -> Palettes (0.0.2)
    - NOTE  | xcodebuild:  note: Using new build system
    - NOTE  | xcodebuild:  note: Building targets in parallel
    - NOTE  | xcodebuild:  note: Using codesigning identity override: -
    - NOTE  | xcodebuild:  note: Build preparation complete
    - NOTE  | [iOS] xcodebuild:  note: Planning build
    - NOTE  | [iOS] xcodebuild:  note: Analyzing workspace
    - NOTE  | [iOS] xcodebuild:  note: Constructing build description

Pods workspace available at `/var/folders/yt/vszldnq143dgzjmmgr_tqs540000gn/T/CocoaPods-Lint-20210914-3981-y3ldra-Palettes/App.xcworkspace` for inspection.

Palettes passed validation.
```

make sure you are using 1.11.2.

Try `pod --version` first to see the output.

**Comment by dnkoutso** [comment]
`gem update cocoapods` also.

**Comment by dnkoutso** [comment]
The podspec:
```ruby

Pod::Spec.new do |spec|

  spec.name         = "Palettes"
  spec.version      = "0.0.2"
  spec.summary      = "A pod to get palettes and assets."

  spec.description  = <<-DESC
                  A pod to get instant and beautiful color palettes.
                   DESC

  spec.homepage     = "https://github.com/penelopearaujo/palettes"
  spec.license       = 'ISC'

  spec.author        = { "PMPA" => "pmpa@cin.ufpe.br" }

  spec.source       = { :git => "https://github.com/penelopearaujo/palettes.git", :tag => "#{spec.version}" }
  spec.source_files  = "  **/*.{swift,h,m}"
  spec.on_demand_resources = { 'roboto' => ['palettes/palettes/Roboto-Regular.ttf'] }

  spec.pod_target_xcconfig = {"PRODUCT_BUNDLE_IDENTIFIER" => "me.pmpa.palettes" ,"DEVELOPMENT_TEAM" => "PMPA"}

  spec.swift_version = '5.0' 
  spec.platform      = :ios, '14.4'

end
```

**Comment by dnkoutso** [comment]
oooh @penelopearaujo I see its `pod trunk push` that is the issue. This is currently blocked by this PR https://github.com/CocoaPods/trunk.cocoapods.org/pull/317

**Comment by dnkoutso** [comment]
@penelopearaujo I will try to land that PR today or tomorrow (PST) and unblock you. Thanks for your patience.

**Comment by penelopearaujo** [comment]
Yes, that's the problem! I was moving code around to check if the issue was local before replying, haha.
I'll wait for the merge, no problems. Thank you! :)
